js/ui/map.js

```diff
@@ -374,27 +374,60 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
      *
      * @param {Point|Array<number>|Array<Point>|Array<Array<number>>} [pointOrBox] Either [x, y] pixel coordinates of a point, or [[x1, y1], [x2, y2]] pixel coordinates of opposite corners of bounding rectangle. Optional: use entire viewport if omitted.
      * @param {Object} params
-     * @param {string|Array<string>} [params.layer] Only return features from a given layer or layers
-     * @param {boolean} [params.includeGeometry=false] If `true`, geometry of features will be included in the results at the expense of a much slower query time.
+     * @param {Array<string>} [params.layers] Only query features from layers with these layer IDs.
      * @param {Array} [params.filter] A mapbox-gl-style-spec filter.
      * @param {featuresCallback} callback function that receives the results
      *
      * @returns {Map} `this`
      *
      * @example
-     * map.queryRenderedFeatures([20, 35], { layer: 'my-layer-name' }, function(err, features) {
+     * map.queryRenderedFeaturesAsync([20, 35], { layers: ['my-layer-name'] }, function(err, features) {
      *   console.log(features);
      * });
      *
      * @example
-     * map.queryRenderedFeatures([[10, 20], [30, 50]], { layer: 'my-layer-name' }, function(err, features) {
+     * map.queryRenderedFeaturesAsync([[10, 20], [30, 50]], { layers: ['my-layer-name'] }, function(err, features) {
      *   console.log(features);
      * });
      */
-    queryRenderedFeatures: function(pointOrBox, params, callback) {
-        if (callback === undefined) {
+    queryRenderedFeaturesAsync: function(pointOrBox, params, callback) {
+        if (!(pointOrBox instanceof Point || Array.isArray(pointOrBox))) {
             callback = params;
             params = pointOrBox;
+            pointOrBox = undefined;
+        }
+        var queryGeometry = this._makeQueryGeometry(pointOrBox);
+        this.style.queryRenderedFeaturesAsync(queryGeometry, params, this._classes, this.transform.zoom, this.transform.angle, callback);
+        return this;
```

Comment:
Can we move these transformations into `Style#queryRenderedFeaturesAsync`? I would like to keep all that logic together and make this a simple proxy.